File: exemplos/BigLogN.cpp
#include <iostream>

using namespace std;

// O tempo de execução cresce de forma logarítmica.
void logN(int n){
    for (int i = 1; i < n; i *= 2){
        cout << i << endl;
    }
}

int main(int argc, char const *argv[])
{
    logN(100);
    return 0;
}

File: bigO/bigLogN.py
# O(log n) faz com que o tempo de execução cresça de forma logarítmica
def bigLogN(n: int) -> int:
    cont = 0
    while n > 1:
        n //= 2
        cont += 1
    return cont

def main():
    print(bigLogN(5))
    print(bigLogN(16))

if __name__ == "__main__":
    main()
